services/m3aggregator/config/aggregator.go

```diff
@@ -531,50 +550,70 @@ func (c *kvClientConfiguration) NewKVClient(instrumentOpts instrument.Options) (
 	return c.Etcd.NewClient(instrumentOpts)
 }
 
-type shardFnType string
+type hashFnType string
 
-// List of supported sharding function types.
+// List of supported hashing function types.
 const (
-	murmur32ShardFn shardFnType = "murmur32"
+	murmur32HashFn hashFnType = "murmur32"
 
-	defaultShardFn = murmur32ShardFn
+	defaultHashFn = murmur32HashFn
 )
 
 var (
-	validShardFnTypes = []shardFnType{
-		murmur32ShardFn,
+	validHashFnTypes = []hashFnType{
+		murmur32HashFn,
 	}
 )
 
-func (t *shardFnType) UnmarshalYAML(unmarshal func(interface{}) error) error {
+func (t *hashFnType) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	var str string
 	if err := unmarshal(&str); err != nil {
 		return err
 	}
 	if str == "" {
-		*t = defaultShardFn
+		*t = defaultHashFn
 		return nil
 	}
-	validTypes := make([]string, 0, len(validShardFnTypes))
-	for _, valid := range validShardFnTypes {
+	validTypes := make([]string, 0, len(validHashFnTypes))
+	for _, valid := range validHashFnTypes {
 		if str == string(valid) {
 			*t = valid
 			return nil
 		}
 		validTypes = append(validTypes, string(valid))
 	}
-	return fmt.Errorf("invalid shrading function type '%s' valid types are: %s",
+	return fmt.Errorf("invalid hashing function type '%s' valid types are: %s",
 		str, strings.Join(validTypes, ", "))
 }
 
-func (t shardFnType) ShardFn() (aggregator.ShardFn, error) {
+func (t hashFnType) ShardFn() (aggregator.ShardFn, error) {
 	switch t {
-	case murmur32ShardFn:
+	case murmur32HashFn:
 		return func(id []byte, numShards int) uint32 {
 			return murmur3.Sum32(id) % uint32(numShards)
 		}, nil
 	default:
-		return nil, fmt.Errorf("unrecognized hash gen type %v", t)
+		return nil, fmt.Errorf("unrecognized shard function type %v", t)
+	}
+}
+
+func (t hashFnType) AggregatedShardFn() (aggregator.AggregatedShardFn, error) {
+	switch t {
+	case murmur32HashFn:
+		// NB(xichen): The function only allocates when the id of the aggregated metric
+		// is more than initialChunkedIDSize (i.e., 256 bytes) in size and requires zero
```

Comment:
super nit: seems `initialChunkedIDSize ` is actually set to 512 above.

Reply:
Updated.